pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -168,18 +169,35 @@ func (r *ReconcileClusterDisruption) reconcilePDBsForOSDs(
 		}
 		return fmt.Errorf("could not check cluster health: %+v", err)
 	}
+	if pdbStateMap.Data == nil {
+		pdbStateMap.Data = make(map[string]string)
+	}
 	_, ok := pdbStateMap.Data[disabledPDBKey]
 	if !ok {
 		pdbStateMap.Data[disabledPDBKey] = ""
 	}
-	if len(drainingFailureDomains) != 0 {
+	var disabledPDBTime time.Time
+	timeString, disabledPDBTimeSet := pdbStateMap.Data[disabledPDBTimeKey]
+	if disabledPDBTimeSet {
+		disabledPDBTime, err = time.Parse(time.RFC3339, timeString)
+		if err != nil {
+			logger.Errorf("Could not parse timestampe %v: %v", disabledPDBTime, err)
```

Comment:
@rohantmp If the time.Parse fails, we still continue. Is the disabledPDBTime defined in this case?

Reply:
No, fixed it.